FreestarAds-Ogury/Source/objc/OguryBannerMediator.m

```diff
@@ -0,0 +1,119 @@
+//
+//  OguryBannerMediator.m
+//  FreestarAds-Ogury
+//
+//  Created by Carlos Alcala on 7/04/22.
+//
+
+#import "OguryBannerMediator.h"
+#import <FreestarAds/FreestarAds.h>
+#import <UIKit/UIKit.h>
+#import <OgurySdk/Ogury.h>
+#import <OguryAds/OguryAds.h>
+#import <OguryChoiceManager/OguryChoiceManager.h>
+
+@interface OguryBannerMediator()<OguryBannerAdDelegate, FSTRMediatorEnabling>
+
+@property (nonatomic, strong) OguryBannerAd *ad;
+@property OguryAdsBannerSize *requestedSize;
+@property CGFloat requestedWidth;
+@property(nonatomic, assign) FreestarBannerAdSize freestarAdSize;
+
+
+@end
+
+@implementation OguryBannerMediator
+
+- (BOOL)isEnabledForMediatorFormat:(FSTRMediatorFormatType)type {
+    switch (type) {
+        case FSTRMediatorFormatTypeBanner:
+            return YES;
+        default:
+            return NO;
+    }
+}
+
+- (BOOL)canShowBannerAd {
+    return YES;
+}
+
+- (BOOL)isAdaptiveEnabled {
+    return [Freestar adaptiveBannerEnabled];
+}
+
+- (NSString*)placementId {
+    return self.mPartner.placement_id;
+}
+
+-(void)loadBannerAd {
+    FSTRLog(@"OGURY: loadBannerAd");
+
+    self.ad = [[OguryBannerAd alloc] initWithAdUnitId:[self placementId]];
+    [self.ad loadWithSize:self.requestedSize];
+    self.ad.delegate = self;
+}
+
+#pragma mark - showing
+
+- (void)showAd {
+    if (self.ad) {
+        FSTRLog(@"OGURY: showAd - placeAdContent");
+        // Place the ad view onto the screen.
+        [self.container placeAdContent:self.ad];
+    }
+}
+
+#pragma mark - OguryBannerAdDelegate
+
+- (void)didLoadOguryBannerAd:(OguryBannerAd *)banner {
+    FSTRLog(@"OGURY: didLoadOguryBannerAd");
+
+    self.ad = banner;
+    [self partnerAdLoaded];
+}
+
+- (void)didClickOguryBannerAd:(OguryBannerAd *)banner {
+    FSTRLog(@"OGURY: didClickOguryBannerAd");
+    [self partnerAdClicked];
+}
+- (void)didCloseOguryBannerAd:(OguryBannerAd *)banner {
+    FSTRLog(@"OGURY: didCloseOguryBannerAd");
+}
+
+- (void)didDisplayOguryBannerAd:(OguryBannerAd *)banner {
+    FSTRLog(@"OGURY: didDisplayOguryBannerAd");
+
+    [self partnerAdShown];
+}
+
+- (void)didFailOguryBannerAdWithError:(OguryError *)error forAd:(OguryBannerAd *)banner {
+    FSTRLog(@"OGURY: didFailOguryBannerAdWithError %@", [error localizedDescription]);
+
+   [self partnerAdLoadFailed:[NSString stringWithFormat:@"%ld", (long)error.code]];
+}
+
+- (void)didTriggerImpressionOguryBannerAd:(OguryBannerAd *)banner {
+    FSTRLog(@"OGURY: didTriggerImpressionOguryBannerAd");
+}
+
+- (UIViewController *)presentingViewControllerForOguryAdsBannerAd:(OguryBannerAd*)banner {
+    return nil;
```

Comment:
Is this correct? I'm just wondering why we have a method that always returns nil.

Reply:
I will fix this @maddapper  it's the same problem we had on Yahoo adapter.